explore_data/nbconverted/viz_recount2_compendium.py

```diff
@@ -0,0 +1,143 @@
+random_state = 123
+seed(random_state)
+
+
+# In[2]:
+
+
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../"))
+
+config_file = os.path.abspath(os.path.join(base_dir,
+                                           "config_human.tsv"))
+params = utils.read_config(config_file)
+
+
+# In[3]:
+
+
+# Load params
+normalized_data_file = params['normalized_compendium_data_file']
+validation_frac = params['validation_frac']
+
+
+# In[4]:
+
+
+# Read data
+normalized_compendium = pd.read_table(
+    normalized_data_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+print(normalized_compendium.shape)
+normalized_compendium.head()
+
+
+# In[5]:
+
+
+# Get validation and training set
+# random_state matches the state used in the training
+test_set_percent = validation_frac
+val_df = normalized_compendium.sample(frac=test_set_percent, random_state=123)
+val_samples = list(val_df.index)
+
+
+# In[6]:
+
+
+# UMAP embedding of original input data
+
+# Get and save model
+model = umap.UMAP(random_state=random_state).fit(normalized_compendium)
+
+input_data_UMAPencoded = model.transform(normalized_compendium)
+input_data_UMAPencoded_df = pd.DataFrame(data=input_data_UMAPencoded,
+                                         index=normalized_compendium.index,
+                                         columns=['1','2'])
+# Add label
+input_data_UMAPencoded_df['dataset'] = 'training'
+input_data_UMAPencoded_df.loc[val_samples,'dataset'] = 'validation'
+
+input_data_UMAPencoded_df
+
+
+# In[12]:
+
+
+# Plot
+fig = ggplot(input_data_UMAPencoded_df, aes(x='1', y='2'))
+fig += geom_point(aes(color='dataset'), alpha=0.2)
+fig += labs(x ='UMAP 1',
+            y = 'UMAP 2',
+            title = 'UMAP of normalized compendium')
+fig += theme_bw()
+fig += theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    legend_title=element_text(family='sans-serif', size=15),
+    legend_text=element_text(family='sans-serif', size=12),
+    plot_title=element_text(family='sans-serif', size=15),
+    axis_text=element_text(family='sans-serif', size=12),
+    axis_title=element_text(family='sans-serif', size=15)
+    )
+fig += guides(colour=guide_legend(override_aes={'alpha': 1}))
+fig += scale_color_manual(['#ff6666', '#add8e6'])
+
+print(fig)
+
+
+# **Observations:**
+# * There looks to be a good amount of variance in the compendium overall.
+# * Using a split of 25% seems to get a similar distribution of data between training and validation sets.
+# * Remember, the dataset is in 17K dimensional space, which will make the small clusters difficult to represent during training
+# 
+# Overall, having so many features in our dataset, points to the need for more samples to represent the structure in the compendium. For now, we are limited by memory to only select a subset of recount2, but in a future iteration perhaps this will be updated.
```

Comment:
In what way are you memory limited? If you're training a VAE, couldn't you use a [Tensorflow Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset) so that you don't have to fit all the data into memory during training?

Reply:
Currently, I am running this on my office desktop and the full dataset used for training is being held in memory so I am limited by the memory on this machine. But I have talked with Casey about training on the full recount2 dataset by reading the data into batches as youre suggested. The downloading of the data from recount2 may also be contributing to the memory issue but I have not fully tested this yet. I will make an issue for the future to address this.